benchopt/utils/datasets/simulated.py

```diff
@@ -0,0 +1,79 @@
+"""Helper to generate simulated data in benchopt."""
+import numpy as np
+from numpy.linalg import norm
+
+from ..checkers import check_random_state
+
+
+def make_correlated_data(n_samples=100, n_features=50, rho=0.6, snr=3,
+                         w_true=None, nnz=10, random_state=None):
```

Comment:
Maybe change the name to `make_correlated_regression`?
Also, I would rather have a `density` parameter I think, not a big fan of having to move this if we change the number of features.

For the generation, what do you think, should we have a fixed number of coef or something that can vary a bit around the "mean" that is density?